sentry/src/main/java/io/sentry/transport/ReusableCountLatch.java

```diff
@@ -0,0 +1,234 @@
+// Apache-2:
+// https://github.com/MatejTymes/JavaFixes/blob/37e74b9d0a29f7a47485c6d1bb1307f01fb93634/LICENSE
+package io.sentry.transport;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.AbstractQueuedSynchronizer;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Class originally copied from <a
+ * href="https://github.com/MatejTymes/JavaFixes/blob/37e74b9d0a29f7a47485c6d1bb1307f01fb93634/src/main/java/javafixes/concurrency/ReusableCountLatch.java">ReusableCountLatch.java</a>.
+ *
+ * <p>A synchronization aid that allows one or more threads to wait until a set of operations being
+ * performed in other threads completes.
+ *
+ * <p>
+ *
+ * <p>A {@code ReusableCountLatch} is initialized with a given <em>count</em>. The {@link
+ * #waitTillZero} methods block until the current count reaches zero due to invocations of the
+ * {@link #decrement} method, after which all waiting threads are released. If zero has been reached
+ * any subsequent invocations of {@link #waitTillZero} return immediately. The coun cen be increased
+ * calling the {@link #increment()} method and any subsequent thread calling the {@link
+ * #waitTillZero} method will be block again until another zero is reached.
+ *
+ * <p>
+ *
+ * <p>{@code ReusableCountLatch} provides more versatility than {@link
+ * java.util.concurrent.CountDownLatch CountDownLatch} as the count doesn't have to be known upfront
+ * and you can reuse this class as many times as you want to. It is also better than a {@link
+ * java.util.concurrent.Phaser Phaser} whose count is limited to 65_535. {@code ReusableCountLatch}
+ * instead can count up to 2_147_483_647 (2^31-1).
+ *
+ * <p>
+ *
+ * <p>Great use case for {@code ReusableCountLatch} is when you wait for tasks on other threads to
+ * finish, but these tasks could trigger more tasks and it is not know upfront how many will be
+ * triggered in total.
+ *
+ * @author mtymes
+ * @since 07/10/16 00:10 AM
+ */
+public final class ReusableCountLatch {
```

Comment:
can we make package private?

Reply:
No, because it's used by all transport implementations.